compiler/qsc_circuit/src/builder.rs

```diff
@@ -252,17 +255,24 @@ impl Builder {
             .unwrap_or_default()
     }
 
-    fn finish_circuit(&self, mut circuit: Circuit) -> Circuit {
+    fn finish_circuit(&self, operations: Vec<Operation>) -> Circuit {
+        let mut qubits = vec![];
+
         // add qubit declarations
         for i in 0..self.remapper.num_qubits() {
             let num_measurements = self.num_measurements_for_qubit(WireId(i));
-            circuit.qubits.push(crate::circuit::Qubit {
+            qubits.push(crate::circuit::Qubit {
                 id: i,
-                num_children: num_measurements,
+                num_results: num_measurements,
             });
         }
 
-        circuit
+        let max_q_id = qubits.iter().map(|qubit| qubit.id).max().unwrap_or(0);
+
+        Circuit {
+            component_grid: op_grid_to_comp_grid(operation_list_to_grid(operations, max_q_id)),
```

Comment:
`op_grid_to_comp_grid(operation_list_to_grid(...))` are always called together like this. Maybe just expose one function, `operation_list_to_component_grid`?

Reply:
I've moved the functionality of `op_grid_to_comp_grid` into `operation_list_to_grid` so there's only one function now, except in the test file where `op_grid_to_comp_grid` was still useful.